openstack/blockstorage/extensions/schedulerhints/requests.go

```diff
@@ -0,0 +1,132 @@
+package schedulerhints
+
+import (
+	"regexp"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+// SchedulerHints represents a set of scheduling hints that are passed to the
+// OpenStack scheduler.
+type SchedulerHints struct {
+	// DifferentHost will place the volume on a different back-end that does not
+	// host the given volumes.
+	DifferentHost []string
+
+	// SameHost will place the volume on a back-end that hosts the given volumes.
+	SameHost []string
+
+	// LocalToInstance will place volume on same host on a given instance
+	LocalToInstance string
+
+	// Query is a conditional statement that results in back-ends able to
+	// host the volume.
+	Query []interface{}
```

Comment:
Why is this `[]interface{}`?

Reply:
Also, we can find `query` used in https://github.com/openstack/cinder/blob/master/cinder/scheduler/filters/json_filter.py

Reply:
But it is defined as a string here: https://github.com/openstack/cinder/blob/b8167a5c3e5952cc52ff8844804b7a5ab36459c8/cinder/api/schemas/scheduler_hints.py#L69

The comments indicate that `jsonutils.loads` is used to convert a string to a dict.

Reply:
OK,Got it. :)